### DISCUSSION TITLE
Tilt axis specification

### DISCUSSION THREAD

**Discussion by asarnow** [nextpyp/.github#21] [Q&A] [answered]
I'm confused about how the initial tilt axis estimate should be specified. The tiltxcorr documentation and the nextPYP help text say the axis is the CCW-postive angle from the Y axis to the tilt axis. On our Krios/K3 it ~85˚ (just shy of horizontal, the long axis of the K3). Fiducial alignment is robust and the right tilt axis was always found, but with fiducial-free TS the wrong axis was being selected until I went through the logs and saw the number passed to tiltxcorr appears to be the nextPYP value minus 90˚. If I give nextPYP 175˚ (so that tiltxcorr gets 85˚) then the correct tilt axis is found. It is displayed as "85˚" in the nextPYP plots.

Have I misunderstood the documentation? Does PYP consider the horizontal axis to be Y?

**Comment by tofushan** [comment, marked as answer]
Hi, 

Thanks for asking. In order to automatically obtain the tilt axis rotation angle from mode files, we decided to extract the value from the field of "RotationAngle" in mdoc files for "Tilt-axis angle (degrees)" in nextPYP's "Import raw data" block. However, it follows different convention from IMOD, according to [SerialEM documentation](https://bio3d.colorado.edu/SerialEM/betaHlp/html/about_formats.htm).

For example, if you manually enter 85 degree in the "Tilt-axis angle (degrees)", nextPYP would subtract 90 from it for tiltxcorr, which is -5 degree. 

I hope it answers your question, and thanks for using nextPYP for your projects.

**Comment by asarnow** [comment]
Thanks for the explanation! Indeed, the .mdoc RotationAngle field has the right value (175˚) as opposed to "tilt axis angle" in the header, and this makes sense as per the quoted SerialEM doc below. However, my confusion really came from the text in the nextPYP interface, "Approximate orientation of tilt-axis with respect to the y-axis" which sounds like the IMOD convention.

NextPYP:
<img width="1002" alt="image" src="https://github.com/nextpyp/.github/assets/4052606/c37f9a7c-6776-4ee8-b270-6c707d3b4e1e">

SerialEM doc:
> Rotation of image from having the tilt axis along its X axis (CCW +).  This corresponds to the definition of the approximate rotation angles reported by Thermo/FEI scripting.  It is different from the 'Tilt axis angle' in the second title of the file header and .mdoc, which follows the IMOD convention of tilt axis rotation relative to the Y axis of the image. Thus, the latter will be 90 degrees less than the RotationAngle in a system with no axis inversions.

**Reply by tofushan** [reply]
Yes, you are right. The description we have in the interface is the IMOD convention. We will correct it in the next release, and thanks for reporting!
